src/realm_dart_sync.cpp

```diff
@@ -25,7 +25,7 @@ RLM_API void realm_dart_http_request_callback(realm_userdata_t userdata, const r
     // the pointers in error are to stack values, we need to make copies and move them into the scheduler invocation
     struct request_copy_buf {
         std::string url;
-        std::string body;
+        std::vector<char> body;
```

Comment:
@nielsenko @fealebenpae can you please look at this logic. It's somewhat convoluted and I'm not sure I understand it correctly, but as far as I understand the docs, `std::string` is not a container, so it cannot be used as a buffer the way we're using it. What I'm seeing is that before and after the `ud->scheduler->invoke` on line 52, `request_copy_buf.body.data()` is returning different addresses if `body` is `std::string`. This is causing `request_copy.body` to point to mangled data. I don't know why we need the map and the vector for the headers and don't know if the url may suffer from similar problems.

Reply:
With the latest change.. reverting to `std::string` for body in `request_copy_buf`, but re-constructing the `request_body` from the moved `buf` inside the callback I agree.

Nice find :-)

Reply:
I suggest we fix it instead of changing to std::vector also we do the same in `realm_dart_sync_error_handler_callback`

Reply:
How do you suggest fixing it?

Reply:
I was going to suggest a fix today. Will write back shortly.

Reply:
Question.. for a while you reverted the `vector<char>` change.. why did you bring it back in b1e6f34a163a72d0996750b56442a61baf6ba2ae?

Reply:
I was investigating failures on CI which turned out to be due to https://github.com/realm/realm-core/issues/5890. Once I found that out, I went back to strings.

Reply:
I did a fix PR here https://github.com/realm/realm-dart/pull/924.

Reply:
The fix in #924 seems to be exactly what I'm doing here - how is it different?

Reply:
It is different since it uses std::string instead of manually handling strings in a std::vactor. It also makes explicit where exactly we copy the values and take pointers on them. If in the future someone needs to add another string to that buffer it is not error prone to do. 
This hear does not fix the original issue of taking pointers to freed memory.

Reply:
I'm not using vectors in this PR? As you can see this comment is outdated because it was made on code that changed. The current implementation is using std::strings with the only difference being that I'm std::move-ing the original request, which is not a problem in and of itself, but I'd be happy to just copy it instead.

Reply:
hm, I did review on a vector version. Let me check the current code. We should try to sync somehow reviews with newer code changes.

Reply:
ok this one is a bit different. If you copy the request and then std::move it it will be the same as #924. I suggest we go with #924 since it already split the fix into a separate entity + it is fixing up the changelog.

Reply:
Yeah, let's merge #924 as there's a lot of commentary by Mathias. I'll rebase this after.